### PR TITLE
Support `normalization_resolver` in `BasicGNN`

### PR DESCRIPTION
**Pull request by lightaime**
This PR added support for `normalization_resolver` in `basic_gnn`.

### CODE REVIEW

File: torch_geometric/nn/models/basic_gnn.py [hunk 1]
```diff
@@ -100,9 +106,19 @@ def __init__(
         if norm is not None:
             self.norms = ModuleList()
             for _ in range(num_layers - 1):
-                self.norms.append(copy.deepcopy(norm))
+                norm_layer = normalization_resolver(
```
Comment (rusty1s) [inline torch_geometric/nn/models/basic_gnn.py:109]: I think we still need to deep copy it here.

File: torch_geometric/nn/models/basic_gnn.py [hunk 2]
```diff
@@ -57,7 +63,8 @@ def __init__(
         out_channels: Optional[int] = None,
         dropout: float = 0.0,
         act: Union[str, Callable, None] = "relu",
-        norm: Optional[torch.nn.Module] = None,
+        norm: Optional[str] = None,
```
Comment (Padarn) [inline torch_geometric/nn/models/basic_gnn.py:66]: Should this be a `Union[str, Callable]`?

**Review by Padarn — APPROVED** [review]

File: torch_geometric/nn/models/basic_gnn.py (hunk 2 above)
Comment (rusty1s) [reply torch_geometric/nn/models/basic_gnn.py:66]: Good catch. Fixed.

**Review by rusty1s — APPROVED** [review]
